app/src/main/java/org/simple/clinic/summary/PatientSummaryEffectHandler.kt

```diff
@@ -68,40 +67,28 @@ class PatientSummaryEffectHandler @AssistedInject constructor(
         .build()
   }
 
-  // TODO(vs): 2020-01-15 Revisit after Mobius migration
   private fun loadPatientSummaryProfile(scheduler: Scheduler): ObservableTransformer<LoadPatientSummaryProfile, PatientSummaryEvent> {
     return ObservableTransformer { effects ->
-      effects.flatMap { fetchPatientSummaryProfile ->
-        val patientUuid = fetchPatientSummaryProfile.patientUuid
-
-        val sharedPatients = patientRepository.patient(patientUuid)
-            .subscribeOn(scheduler)
-            .map {
-              // We do not expect the patient to get deleted while this screen is already open.
-              (it as Just).value
-            }
-            .replay(1)
-            .refCount()
-
-        val addresses = sharedPatients
-            .flatMap { patient -> patientRepository.address(patient.addressUuid).subscribeOn(scheduler) }
-            .map { (it as Just).value }
-
-        val latestPhoneNumberStream = patientRepository.phoneNumber(patientUuid).subscribeOn(scheduler)
-        val latestBpPassportStream = patientRepository.bpPassportForPatient(patientUuid).subscribeOn(scheduler)
-        val bangladeshNationalIdStream = patientRepository.bangladeshNationalIdForPatient(patientUuid).subscribeOn(scheduler)
-
-        Observables
-            .combineLatest(sharedPatients, addresses, latestPhoneNumberStream, latestBpPassportStream, bangladeshNationalIdStream) { patient, address, phoneNumber, bpPassport, bangladeshNationalId ->
-              PatientSummaryProfile(patient, address, phoneNumber.toNullable(), bpPassport.toNullable(), bangladeshNationalId.toNullable())
-            }
-            .take(1)
-            .map(::PatientSummaryProfileLoaded)
-            .cast<PatientSummaryEvent>()
-      }
+      effects
+          .observeOn(scheduler)
+          .map { patientRepository.patientProfileImmediate(it.patientUuid) }
+          .filterAndUnwrapJust()
+          .map { it.withoutDeletedBusinessIds().withoutDeletedPhoneNumbers() }
```

Comment:
Do we need to filter this here or we can do it at the query level? 
In what scenario will we need deleted values?

Reply:
The scenario in which we need deleted values is because this specific query is also used for fetching data for syncing, in which case we should not exclude deleted values.

One option is that we can add an alternate repository method that lets us drop these records at the query level, but I thought that this PR was quite large enough already without adding it.

This is why I wanted to add this here first and see what someone else has to say about it.

Reply:
I would recommend using either a query to filter this out or the repository would make more sense. But this is alright for now. I just wanted to know the reason.

Reply:
Let's take a look at the rest of the screens if we need this model (`PatientProfile`) there and if yes, we'll add this as a separate repository method.